src/Crypto/RNCryptor/Types.hs

```diff
@@ -70,9 +73,21 @@ randomSaltIO :: Int -> IO ByteString
 randomSaltIO sz = C8.pack <$> forM [1 .. sz] (const $ randomRIO ('\NUL', '\255'))
 
 --------------------------------------------------------------------------------
+makeKey :: ByteString -> ByteString -> ByteString
+makeKey = generate (prfHMAC (undefined::SHA1)) (Parameters 10000 32) -- userKey salt
+
+--------------------------------------------------------------------------------
+makeHMAC :: ByteString -> ByteString -> ByteString -> ByteString
+makeHMAC hmacSalt userKey secret =
+  let key        = makeKey userKey hmacSalt::ByteString
```

Comment:
You don't need the type annotation on `hmacSalt`, as the compiler knows it's a `ByteString` just by looking at the type of the first argument.